api/src/main/java/org/openmrs/module/xdssender/api/xds/MessageUtil.java

```diff
@@ -125,8 +139,151 @@ private ProvideAndRegisterDocumentSetRequestType addDocToRequest(ProvideAndRegis
         return requestType;
     }
 
-	private SubmitObjectsRequest createClinicalSubmitRequest(DocumentInfo info,
-                                Encounter encounter, ExtrinsicObjectType oddRegistryObject) throws JAXBException  {
+    private SubmitObjectsRequest addToRegistryRequest(SubmitObjectsRequest registryRequest, DocumentData docData,
```

Comment:
why did you have add so much code? What does it do exactly?

Reply:
I've:
* Splited old createProvideAndRegisterDocument method into:
createClinicalSubmitRequest and createClinicalExtrinsicObjectType
(logic responsible for generating CDA stays the same)

* Refactored logic of createProvideAndRegisterDocument by spliting it's
logic into several methods (used for generating the HL7 attachment)

createClinicalSubmitRequest and createClinicalExtrinsicObjectType can be relaplaced with new metody aftherwards